kafka-impl/src/main/java/io/streamnative/pulsar/handlers/kop/KafkaCommandDecoder.java

```diff
@@ -85,9 +82,20 @@ public void exceptionCaught(ChannelHandlerContext ctx, Throwable cause) {
     }
 
     protected void close() {
+        assert !isActive.get();
```

Comment:
if assert failed, it will throw exception?

Reply:
No. Expect we enabled `-ea` option. `assert` is used for logic assumption. Because we want to make sure `KafkaCommandDecoder#close` is called in the block of `if (isActive.getAndSet(false))`, which means `isActive.get()` must return false expect there's a bug with Java standard library.

But I'll remove this line. Because I used `assert` here to ensure `isActive` is false and when `writeAndFlushResponseToClient` is called by null completion, then the loop will be skipped. However, we use `cancel` now so that `writeAndFlushResponseToClient` won't be called and the assertion becomes useless.